internal/namespaces/object/v1/custom_bucket.go

```diff
@@ -0,0 +1,597 @@
+	}
+	bucket.Owner = normalizeOwnerID(acl.Owner.ID)
+	bucket.ACL = awsACLToCustomGrants(acl)
+
+	// get endpoints
+	bucket.APIEndpoint = getAPIEndpoint(region.String())
+	bucket.BucketEndpoint, err = getBucketEndpoint(name, region.String())
+	if err != nil {
+		return nil, err
+	}
+
+	return bucket, nil
+}
+
+func getAPIEndpoint(region string) string {
+	if customEndpoint := os.Getenv("SCW_S3_ENDPOINT"); customEndpoint != "" {
+		return customEndpoint
+	}
+	return fmt.Sprintf("https://s3.%s.scw.cloud", region)
+}
+
+func getBucketEndpoint(name, region string) (string, error) {
+	if customEndpoint := os.Getenv("SCW_S3_ENDPOINT"); customEndpoint != "" {
+		u, err := url.Parse(customEndpoint)
+		if err != nil {
+			return "", fmt.Errorf("could not parse custom endpoint %s: %w", customEndpoint, err)
+		}
+		u = u.JoinPath(name, u.Path)
+		return u.String(), nil
+	}
+	return fmt.Sprintf("https://%s.s3.%s.scw.cloud", name, region), nil
+}
+
+func countBucketObjects(ctx context.Context, client *s3.Client, name string) (nbObjects, nbParts int64, totalSize scw.Size, err error) {
+	var size int64
+
+	// count full objects
+	objectsList, err := client.ListObjectsV2(ctx, &s3.ListObjectsV2Input{
+		Bucket: &name,
+	})
+	if err != nil {
+		return nbObjects, nbParts, totalSize, fmt.Errorf("could not list objects: %w", err)
+	}
+	for _, object := range objectsList.Contents {
+		nbObjects++
+		size += *object.Size
+	}
+
+	// count parts
+	multipartUploads, err := client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{
+		Bucket: &name,
+	})
+	if err != nil {
+		return nbObjects, nbParts, totalSize, fmt.Errorf("could not list multipart uploads: %w", err)
+	}
+	for _, upload := range multipartUploads.Uploads {
+		partsList, err := client.ListParts(ctx, &s3.ListPartsInput{
+			Bucket:   &name,
+			Key:      upload.Key,
+			UploadId: upload.UploadId,
+		})
+		if err != nil {
+			return nbObjects, nbParts, totalSize, fmt.Errorf("could not list parts: %w", err)
+		}
+		for _, part := range partsList.Parts {
+			nbParts++
+			size += *part.Size
+		}
+	}
+
+	return nbObjects, nbParts, scw.Size(size), nil
+}
+
+func putBucketVersioning(ctx context.Context, client *s3.Client, name string, enabled bool) error {
+	request := &s3.PutBucketVersioningInput{
+		Bucket: &name,
+	}
+	if enabled {
+		request.VersioningConfiguration = &types.VersioningConfiguration{
+			Status: types.BucketVersioningStatusEnabled,
+		}
+	} else {
+		request.VersioningConfiguration = &types.VersioningConfiguration{
+			Status: types.BucketVersioningStatusSuspended,
+		}
+	}
+	_, err := client.PutBucketVersioning(ctx, request)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func putBucketTagging(ctx context.Context, client *s3.Client, name string, tags []string) error {
+	if len(tags) == 0 {
+		return nil
+	}
+	newTags := []types.Tag(nil)
+	for _, tag := range tags {
+		trim := strings.Trim(tag, "\"")
+		split := strings.Split(trim, "=")
```

Comment:
It seems strange to have a split here while we could have tags.foo=bar

Reply:
Plus we could have completion on tags.foo for update in case we fetch all the tags.